Applications/Spire/Source/OrderImbalanceIndicator/FilteredOrderImbalanceIndicatorModel.cpp

```diff
@@ -0,0 +1,122 @@
+#include "Spire/OrderImbalanceIndicator/FilteredOrderImbalanceIndicatorModel.hpp"
+#include "Nexus/Definitions/Market.hpp"
+#include "Spire/Spire/QtPromise.hpp"
+
+using Filter = Spire::FilteredOrderImbalanceIndicatorModel::Filter;
+using namespace Nexus;
+using namespace Spire;
+
+FilteredOrderImbalanceIndicatorModel::FilteredOrderImbalanceIndicatorModel(
+    std::shared_ptr<OrderImbalanceIndicatorModel> source_model,
+    std::vector<Filter> filters)
+    : m_source_model(std::move(source_model)),
+      m_filters(std::move(filters)) {}
+
+OrderImbalanceIndicatorModel::SubscriptionResult
+    FilteredOrderImbalanceIndicatorModel::subscribe(
+    const boost::posix_time::ptime& start,
+    const boost::posix_time::ptime& end,
+    const OrderImbalanceSignal::slot_type& slot) {
+  auto signal = std::make_shared<Subscription>(start, end);
+  auto callback = [=] (const auto& imbalance) {
+    if(is_imbalance_accepted(imbalance)) {
+      signal->m_imbalance_signal(imbalance);
+    }
+  };
+  auto [connection, promise] = m_source_model->subscribe(start, end,
```

Comment:
This class does not need to manage connections, there is a direct 1-1 signal/slot relationship. Instead reuse the signal/slot relationship created by ```m_source_model``` instead of duplicating all of the work.